File: BOJ/11723.rs
use std::io::*;

fn main() {
    let n: usize = {
        let mut input = String::new();
        stdin().read_line(&mut input).unwrap();

        input.trim().parse().unwrap()
    };

    let mut mask = 0usize;
    let mut output = BufWriter::new(stdout());

    (0..n).for_each(|_| {
        let mut input = String::new();
        stdin().read_line(&mut input).unwrap();

        let query: Vec<&str> = input.split_ascii_whitespace().collect();

        let read_int = || -> usize { query[1].parse().unwrap() };

        match query[0] {
            "add" => mask |= 1 << read_int(),
            "remove" => mask &= !(1 << read_int()),
            "check" => {
                let contains = (mask & (1 << read_int()) != 0) as usize;
                writeln!(output, "{contains}").unwrap();
            }
            "toggle" => mask ^= 1 << read_int(),
            "all" => mask = (1 << 21) - 1,
            "empty" => mask = 0,
            _ => {}
        };
    });
}

File: BOJ/11723.cpp
#include <iostream>
#include <array>

using namespace std;

int main() {
  ios_base::sync_with_stdio(false);
  cin.tie(nullptr);

  array<bool, 21> arr{};

  int m;
  cin >> m;

  while (m--) {
    string op;
    cin >> op;
    if (op == "add") {
      int num;
      cin >> num;
      arr[num] = true;
    } else if (op == "remove") {
      int num;
      cin >> num;
      arr[num] = false;
    } else if (op == "check") {
      int num;
      cin >> num;
      cout << arr[num] << '\n';
    } else if (op == "toggle") {
      int num;
      cin >> num;
      arr[num] ^= true;
    } else if (op == "all") {
      arr.fill(true);
    } else { // empty
      arr.fill(false);
    }
  }

  return 0;
}
